extensions/build-gradle/main.ts

```diff
@@ -0,0 +1,366 @@
+        return;
+    }
+
+    // Convert the Uint8 array to ascii
+    let ret = new TextDecoder().decode(gradleResp);
+
+    // TODO: Check for failure?
+
+    ret = ret.split(/\r?\n/)
+              .map(line => line.match(/^[\+\\-]{2,}\sProject\s'(:.*)'/))
+              .filter(line => line && line.length > 0)
+              .map(line => line[1]);
+
+    logSuccess(`Found ${ret.length} additional subprojects`);
+    return ret;
+}
+
+/**
+ *  Runs the `gradle` binary to produce the `gradle-dependencies.txt` file. This
+ *  should do the dependency resolution for this machine.
+ *
+ *  We perform a rudimentary check to make sure there wasn't an outright build
+ *  failure.
+ *
+ *  If a `subproject` is provided, attempts to resolve the dependencies for the
+ *  specified subproject.
+ */
+async function generateGradleDeps(subproject: string) {
+    let cmd = "dependencies";
+
+    if(subproject) {
+        cmd = subproject + ":dependencies";
+    }
+
+    let gradleResp = await invokeGradle(["-q", cmd]);
+
+    if (!gradleResp) {
+        return;
+    }
+
+    // Convert the Uint8 array to ascii
+    const ret = new TextDecoder().decode(gradleResp);
+
+    if(ret.indexOf("BUILD FAILED") > 0) {
+        return;
+    }
+
+    return ret;
+}
+
+async function invokeGradle(cmd) {
+    //Try our directory
+    try {
+        return await Deno.run({
+            cmd: ["./gradlew"].concat(cmd),
+            stdout: "piped"
+        }).output();
+    } catch(e) {
+        //doNothing()
+    }
+
+    //Try single parent directory
+    try {
+        return await Deno.run({
+            cmd: ["../gradlew"].concat(cmd),
+            stdout: "piped"
+        }).output();
+    } catch(e) {
+        //doNothing()
+    }
+
+    //Lastly try using gradle directly
+    try {
+        return await Deno.run({
+            cmd: ["gradle"].concat(cmd),
+            stdout: "piped"
+        }).output();
+    } catch(e) {
+        //doNothing()
+    }
+
+    logError("ERROR: It doesn't look like you have `gradle` installed or " +
+             "gradle wrapper in the current or parent directories");
+}
+
+/**
+ *  Parse a `build.gradle` file and returned the identified dependencies. 
+ */
+async function getBuildGradleDeps(subproject: string) {
+    const gradleDeps = await generateGradleDeps(subproject);
+
+    if(!gradleDeps) {
+        logError("ERROR: Failed to parse dependencies. Check your " +
+                      "`build.gradle` file."); 
+        return;
+    }
+
+    // Parse the dependencies from this file.
+    let foundDeps = parseGradleFile(gradleDeps); 
+
+    return foundDeps;
+}
+
+/**
+ *  Submit the provided dependencies to Phylum for analysis.
+ */
+async function submit(pkgs: object[], project: string, group: string) {
+    if(!pkgs.length) {
+        return;
+    }
+
+    logSuccess("Submitting to Phylum for analysis...");
+        
+    if(!project) {
+        logSuccess("Please specify a project with --project <projectName>");
+    }
```

Comment:
`PhylumApi.analyze` accepts `null` for the project and will pull it from the `.phylum_project` file. This extension should support that.